<launch>
    <arg name="planner_type" default="rrt_star"/>

    <arg name="world_frame" default="map"/> 
    <arg name="ugv_frame" default="ugv_base_link"/> 
    
    <arg name="distance_obstacle_ugv"    default="0.5"/>   

    <!-- To wirte Files -->
    <arg name="map_file"        default="agriculture"/>
    <arg name="files_results"    default="results_marsupial_optimizer"/>     
    <!-- RRT Parameters -->
    <arg name="n_iter_rrt"          default="2000"/>
    <arg name="n_loop_rrt"          default="10"/>
    <arg name="radius_near_nodes"   default="1.2"/> 
    <arg name="goal_gap_m"          default="0.2"/> # 1.4
    <arg name="step_steer"          default="0.5"/>
    <arg name="min_l_steer_ugv"     default="8.0"/>
    <arg name="samp_goal_rate"      default="5"/>  <!-- sample : 15-->
    <arg name="sample_mode_rrt"     default="1"/>  <!-- Values: 1= Priority is move UAV >> UAV & UGV >> UGV ,  2= Priority is move UAV & UGV >> UAV >> UGV >>  -->

    <!-- Global Planner: RRT* -->
    <node pkg="planners_pkg" type="global_planner_node" name="planner_node" output="screen"># launch-prefix="gdb -ex run --args">
        <rosparam command="load"                    file="$(find planners_pkg)/cfg/$(arg map_file)_ws.yaml"/>        
        <param name="path"                          value="$(env HOME)/$(arg files_results)/"/>  
        <param name="planner_type"                  value="$(arg planner_type)"/> 
        <param name="world_frame"                   value="$(arg world_frame)"/> 
        <param name="ugv_base_frame"                value="$(arg ugv_frame)"/> 
        <param name="showConfig"                    value="true"/>
        <param name="n_iter"                        value="$(arg n_iter_rrt)"/>   
        <param name="n_loop"                        value="$(arg n_loop_rrt)"/>   
        <param name="radius_near_nodes"             value="$(arg radius_near_nodes)"/>   
        <param name="step_steer"                    value="$(arg step_steer)"/>   
        <param name="min_l_steer_ugv"               value="$(arg min_l_steer_ugv)"/>   
        <param name="goal_gap_m"                    value="$(arg goal_gap_m)"/>    
        <param name="samp_goal_rate"                value="$(arg samp_goal_rate)"/>  
        <param name="sample_mode"                   value="$(arg sample_mode_rrt)"/>    
        <param name="min_distance_add_new_point"    value="1.0"/>  
        <param name="distance_obstacle_ugv"         value="$(arg distance_obstacle_ugv)"/> # bound distance UGV to obstacles 0.05[m] resolution of the map

        <param name="w_nearest_ugv"                 value="12"/>   # catenary Obstacles collision 0.00100
        <param name="w_nearest_smooth"              value="10"/>   # catenary Z collision 0.00105
        <!-- For Grid3D -->     
        <param name="map_path"                      value="$(find planners_pkg)/maps/$(arg map_file)_obstacles.bt"/>
        <param name="map_file"                      value="$(arg map_file)_obstacles"/>
        <param name="global_frame_id"               value="map"/>
    </node>


    </launch>